champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/assassin/passives/Endurance.java

```diff
@@ -0,0 +1,104 @@
+package me.mykindos.betterpvp.champions.champions.skills.skills.assassin.passives;
+
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import me.mykindos.betterpvp.champions.Champions;
+import me.mykindos.betterpvp.champions.champions.ChampionsManager;
+import me.mykindos.betterpvp.champions.champions.skills.Skill;
+import me.mykindos.betterpvp.champions.champions.skills.types.CooldownSkill;
+import me.mykindos.betterpvp.champions.champions.skills.types.ToggleSkill;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import org.bukkit.Bukkit;
+import org.bukkit.Effect;
+import org.bukkit.Material;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+import org.bukkit.attribute.Attribute;
+import org.bukkit.attribute.AttributeInstance;
+
+@Singleton
+@BPvPListener
+public class Endurance extends Skill implements ToggleSkill, CooldownSkill{
+
+    private double effectDuration;
+    @Inject
+    public Endurance(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+
+    @Override
+    public String getName() {
+        return "Endurance";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+
+        return new String[]{
+                "Drop your Sword / Axe to activate",
+                "",
+                "Instantly gain <stat>3</stat> extra hearts and <effect>Speed III</effect> for <val>" + (5 + (level - 1)) + "</val> seconds",
+                "",
+                "Cooldown: <val>" + getCooldown(level)
+        };
+    }
+
+    @Override
+    public void toggle(Player player, int level) {
+
+        player.addPotionEffect(new PotionEffect(PotionEffectType.SPEED, ((5 + (level - 1)) * 20), 2, true, true));
+        player.getWorld().playSound(player.getLocation(), Sound.ENTITY_WARDEN_DEATH, 1.0F, 1.0F);
+        player.getWorld().playEffect(player.getLocation(), Effect.STEP_SOUND, Material.REDSTONE_BLOCK);
+
+        AttributeInstance maxHealth = player.getAttribute(Attribute.GENERIC_MAX_HEALTH);
+        if (maxHealth != null) {
+            maxHealth.setBaseValue(26.0);
+            if ((player.getHealth()+6) > 26.0) {
+                player.setHealth(26);
+            }
+            else{
+                player.setHealth(player.getHealth()+6);
+            }
+        }
```

Comment:
Not a good idea

Just give them their health back, instead of letting them go over standard max health.

Reply:
Thats kinda the point of the ability. Not enough testing has been done to see if its too strong or not so depending on the results of testing ill change stuff.

Reply:
Setting the maximum health this way will almost certainly cause issues

Take a look at DefensiveAura

Reply:
I did look at defensive aura but health boost cannot give you 3 hearts only 2 and 4. I ran into no bugs while testing is there a specific thing that would break by setting the max health like this?

Reply:
The bugs wouldn't be with the skill, but there could be scenarios where the max health gets stuck at 26 even after they dequip the set etc.

I'd be more inclined to give them 4 hearts but increase the cooldown slightly

Reply:
I did run into those issues but I fixed all the ones I could find. Ill try 4 hearts and increased CD.

Reply:
Could do increased CD, decreased duration, lots of other variables

Its the issues you couldn't find that are the problem